src/anniversary/sf33rd/AcrSDK/ps2/ps2PAD.c

```diff
@@ -47,14 +50,68 @@ static void update_pad_stick_dir(PAD_STICK *st, s16 depth);
 static u8 lever_analog_to_digital(PAD_STICK *st);
 static void PADReadSub(s32 i);
 s32 flPADShockSet(s32 pad_id, u32 level, u32 time);
+static s32 PADDeviceInit();
+static void ps2PADWorkClear();
+static void PADPortOpen(s32 port, s32 slot, PS2Slot *adrs);
+static void PADDeviceDestroy();
+
+s32 flPS2PADModuleInit() {
+    flPS2IopModuleLoad("cdrom0:\\THIRD\\IOP\\MODULES\\DBCMAN.IRX;1", 0, NULL, 0);
+    flPS2IopModuleLoad("cdrom0:\\THIRD\\IOP\\MODULES\\SIO2D.IRX;1", 0, NULL, 0);
+    flPS2IopModuleLoad("cdrom0:\\THIRD\\IOP\\MODULES\\PAD2\\DS2U_D.IRX;1", 0, NULL, 0);
+    return 1;
+}
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/AcrSDK/ps2/ps2PAD", flPS2PADModuleInit);
+s32 tarPADInit() {
+    s32 i;
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/AcrSDK/ps2/ps2PAD", tarPADInit);
+    if (PADDeviceInit() == 0) {
+        return 0;
+    }
+    ps2PADWorkClear();
+    ps2pad_clear.pad_buffer[0] = 0xFF;
+    ps2pad_clear.ix.sw = 0xFFFF;
+    for (i = 0; i < 2; i++) {
+        ps2pad_backup[i] = ps2pad_clear;
+        ps2slot[i].state = 0;
+        ps2slot[i].phase = 0;
+        ps2slot[i].port = 0;
+        ps2slot[i].slot = 0;
+        ps2slot[i].vib = 0;
+        ps2slot[i].buff = (u64 *)pad_dma_buf[i];
+    }
+    if (MtapPort == -1) {
+        PADPortOpen(0, 0, ps2slot);
+        PADPortOpen(1, 0, &ps2slot[1]);
+    } else {
+        for (i = 0; i < MtapSlotMax; i++) {
+            PADPortOpen(MtapPort, i, &ps2slot[i]);
+        }
+        if (MtapSlotMax < 2) {
+            PADPortOpen((MtapPort + 1) & 1, 0, &ps2slot[i]);
+        }
+    }
+
+    for (i = 0; i < 2; i++) {
+        ps2pad_config[i] = ps2PadShotConf_Basic;
+        tarpad_root[i].conn.gc.vib = ps2slot[i].port;
+        tarpad_root[i].conn.gc.etc0 = ps2slot[i].slot;
+        flPadFixedAnalogSelectSwitch[i] = flPadFASS[i] = 0 & 0xFF;
```

Comment:
Is this & necessary?

```suggestion
        flPadFixedAnalogSelectSwitch[i] = flPadFASS[i] = 0;
```